modules/lang-painless/build.gradle

```diff
@@ -218,3 +218,69 @@ tasks.register("regen") {
     }
   }
 }
+
+/**********************************************
+ *         Suggest lexer regeneration          *
+ **********************************************/
+
+configurations {
+    suggestRegenerate
+}
+
+dependencies {
+    regenerate 'org.antlr:antlr4:4.5.3'
+}
+
+String suggestGrammarPath = 'src/main/antlr'
+String suggestOutputPath = 'src/main/java/org/elasticsearch/painless/antlr'
+
+tasks.register("cleanSuggestGenerated", Delete) {
+    delete fileTree(grammarPath) {
+        include 'SuggestLexer.tokens'
+    }
+    delete fileTree(outputPath) {
+        include 'Suggest*.java'
+    }
+}
+
+tasks.register("regenSuggestLexer", JavaExec) {
```

Comment:
When I run this path I'm getting some changes to files and untracked files.

```
git status
On branch jack-autolexer
Your branch is up to date with 'jack/autolexer'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        deleted:    src/main/antlr/SuggestLexer.tokens
        modified:   src/main/java/org/elasticsearch/painless/antlr/SuggestLexer.java

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        src/main/java/org/elasticsearch/painless/antlr/SuggestLexer.tokens
```

Reply:
Discussed with @stu-elastic and the regenSuggestLexer command is meant to be a dependency of regenSuggest which does not leave behind these additional files.